sparkle/src/components/DropdownMenu.tsx

```diff
@@ -102,135 +103,153 @@ export interface DropdownButtonProps {
   className?: string;
   disabled?: boolean;
   children?: React.ReactNode;
+  ref?: React.Ref<HTMLButtonElement>;
+  onClick?: () => void;
 }
 
-DropdownMenu.Button = function ({
-  label,
-  type = "menu",
-  size = "sm",
-  tooltip,
-  icon,
-  children,
-  tooltipPosition = "above",
-  className = "",
-  disabled = false,
-}: DropdownButtonProps) {
-  const finalLabelClasses = classNames(
-    labelClasses.base,
-    labelSizeClasses[size],
-    labelClasses.dark.base,
-    !disabled ? labelClasses.active : "",
-    !disabled ? labelClasses.dark.active : "",
-    !disabled ? labelClasses.hover : "",
-    !disabled ? labelClasses.dark.hover : "",
-    disabled ? labelClasses.disabled : ""
-  );
+DropdownMenu.Button = forwardRef<HTMLButtonElement, DropdownButtonProps>(
+  function DropdownMenuButton(
+    {
+      label,
+      type = "menu",
+      size = "sm",
+      tooltip,
+      icon,
+      children,
+      tooltipPosition = "above",
+      className = "",
+      disabled = false,
+      onClick,
+    },
+    ref
+  ) {
+    const finalLabelClasses = classNames(
+      labelClasses.base,
+      labelSizeClasses[size],
+      labelClasses.dark.base,
+      !disabled ? labelClasses.active : "",
+      !disabled ? labelClasses.dark.active : "",
+      !disabled ? labelClasses.hover : "",
+      !disabled ? labelClasses.dark.hover : "",
+      disabled ? labelClasses.disabled : ""
+    );
 
-  const finalIconClasses = classNames(
-    iconClasses.base,
-    iconClasses.dark.base,
-    !disabled ? iconClasses.hover : "",
-    !disabled ? iconClasses.dark.hover : "",
-    disabled ? iconClasses.disabled : ""
-  );
+    const finalIconClasses = classNames(
+      iconClasses.base,
+      iconClasses.dark.base,
+      !disabled ? iconClasses.hover : "",
+      !disabled ? iconClasses.dark.hover : "",
+      disabled ? iconClasses.disabled : ""
+    );
 
-  const finalChevronClasses = classNames(
-    chevronClasses.base,
-    !disabled ? chevronClasses.hover : "",
-    chevronClasses.dark.base,
-    !disabled ? chevronClasses.dark.hover : "",
-    disabled ? chevronClasses.disabled : ""
-  );
+    const finalChevronClasses = classNames(
+      chevronClasses.base,
+      !disabled ? chevronClasses.hover : "",
+      chevronClasses.dark.base,
+      !disabled ? chevronClasses.dark.hover : "",
+      disabled ? chevronClasses.disabled : ""
+    );
 
-  const buttonRef = useContext(ButtonRefContext);
+    const buttonRef = useContext(ButtonRefContext);
+
+    if (children) {
+      return (
+        <Menu.Button
+          as="div"
+          disabled={disabled}
+          ref={ref ?? buttonRef}
```

Comment:
what happens to buttonRef ? We still need it to be set even if a ref is passed ?

Reply:
I would set if no ref has been passed, no?

Reply:
@tdraier am I overlooking something here?

Reply:
yes but it's not set anymore if a ref is passed ?

Reply:
Oh okay, think I got it. Any better now?